### DISCUSSION TITLE
How can we use backrest to backup docker volumes? (Restic in docker?)

### DISCUSSION THREAD

**Discussion by gerardnico** [garethgeorge/backrest#329] [Q&A]
Hallo,

I'm  actually using [restic in docker](https://restic.readthedocs.io/en/latest/020_installation.html#docker-container) to backup my docker volumes.

The command looks like that:
```
docker run --rm \
--env-file secret.env \
--mount source=<volume-name>,target=<target> \
restic/restic \
backup -H containerName
<target>
```

Is it actually possible to change the command executed?

ie in place of the `restic`, using something like that?
```
docker run --rm \
--env-file secret.env \
--mount source=<volume-name>,target=<target> \
restic/restic \
```

In the [overview](https://garethgeorge.github.io/backrest/docs/operations#overview), is it specified how we can change the binary but not the command. 

If not, is there another way to backup volumes with backrest without making a tar file into the host?

Thanks
Nico

**Comment by garethgeorge** [comment]
Backrest provides a docker image, can you create a single docker container running backrest that you mount all volumes you want to backup into?

**Reply by gerardnico** [reply]
Thanks.

Changing the restic command line may work. I was just checking what was the best way to backup docker volumes.

I'm checking a docker setup on a big machine vs a minimal kubernetes set up with velero.

Have a nice sunday!

**Comment by the-Jack-D** [comment]
I'm running Backrest in a docker container. All of the folders I want to back up are subfolders of a single one that I mounted as read-only in my Backrest Docker container. I just point to what I want to back up that way.
